test/schema/self-test.test.ts

```diff
@@ -0,0 +1,20 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as jsonschema from 'jsonschema';
+import { Testing } from '../util';
+
+test('generated schema is valid JSON Schema draft-04', async () => {
+  // GIVEN
+  const result = jsonschema.validate(
+    Testing.schema,
+    JSON.parse(
+      fs
+        .readFileSync(path.join(__dirname, 'json-schema-draft-04.json'))
```

Comment:
Incidentally, there are newer versions. Not sure how important this is, but should we move to the latest one?

Reply:
Yes, I don't know if there was a reason for choosing this one. 
Saying that, the [validator lib](https://github.com/tdegrunt/jsonschema) we currently use only supports up to draft-07